Hw 9

NZarina opened the pull request:
hw-9 task-1
![Code_PV2x98zdnm](https://github.com/user-attachments/assets/d73ff203-985a-4439-975a-aaf10ac09974)

hw-9 task-2
![image](https://github.com/user-attachments/assets/34c14275-acb3-4d33-8fe6-46e3290c8606)

VladimirLQA committed:
Update task-1.mjs

VladimirLQA committed:
Update task-1.test.js

NZarina committed:
Merge branch 'main' of https://github.com/VladimirLQA/aqa_course_hw_unit_tests

NZarina committed:
Add HW-9

VladimirLQA commented on homework-solutions/js/hw-9/task-2.mjs:
```diff
@@ -14,23 +14,49 @@ const characters = [
+  if (typeof name !== 'string') {
+    console.log('Введите имя заменяемого персонажа в формате string')
+  }  
+  if (typeof newCharacter !== 'object') {
```
`typeof` от `null` тоже `object`

NZarina committed:
Update HW-9 task-2

Updated the updateCharacter function, changed the second if condition adding '|| newCharacter == null'
